examples/api/test/material/snack_bar/snack_bar.2_test.dart

```diff
@@ -0,0 +1,150 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:flutter_api_samples/material/snack_bar/snack_bar.2.dart' as example;
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  testWidgets('Shows correct static elements', (WidgetTester tester) async {
+    await tester.pumpWidget(
+      const example.SnackBarExampleApp(),
+    );
+
+    expect(find.byType(SnackBar), findsNothing);
+    expect(find.widgetWithText(AppBar, 'SnackBar Sample'), findsOneWidget);
+    expect(find.widgetWithText(FloatingActionButton, 'Show Snackbar'), findsOneWidget);
+    expect(find.text('Behavior'), findsOneWidget);
+    expect(find.text('Fixed'), findsOneWidget);
+    expect(find.text('Floating'), findsOneWidget);
+    expect(find.text('Content'), findsOneWidget);
+    expect(find.widgetWithText(SwitchListTile, 'Include close Icon'), findsOneWidget);
+    expect(find.widgetWithText(SwitchListTile, 'Multi Line Text'), findsOneWidget);
+    expect(find.widgetWithText(SwitchListTile, 'Include Action'), findsOneWidget);
+    expect(find.widgetWithText(SwitchListTile, 'Long Action Label'), findsOneWidget);
+
+    await tester.scrollUntilVisible(find.byType(Slider), 30);
+    expect(find.text('Action new-line overflow threshold'), findsOneWidget);
+    expect(find.byType(Slider), findsOneWidget);
+
+  });
+
+  testWidgets('Applies configurations to snackbar', (WidgetTester tester) async {
```

Comment:
This seems unnecessarily long, we could remove `findsNothing` and check for SnackBar content text for correct configuration.

Reply:
This does indeed get quite long, which I didn't notice while writing. The checks are intentionally verbose for the sake of consistency. 

If you think that is problematic I can of course change it.

Reply:
Since you have written quite a few tests, this could be a good opportunity to write a concise test. If you feel stuck, feel free to ask.

Reply:
I removed the `findsNothing` instances that are implied from context. Maybe you could take a look at how that reads?

I personally find that quite hard to read (maybe it's just because configuration is so verbose). Would a designated function inside the test be over the top?
```dart
  testWidgets('Applies configurations to snackbar', (WidgetTester tester) async {
    bool currentCloseIcon = true;
    bool currentMultiLineText = false;
    Future<void> applyConfig({
      SnackBarBehavior? behavior,
      bool? closeIcon,
      bool? multiLineText,
      ...
    }) async {
      switch(behavior) {
        case SnackBarBehavior.fixed:
          await tester.tap(find.text('Fixed'));
        case SnackBarBehavior.floating:
          await tester.tap(find.text('Floating'));
        case null:
          // ignore
      }
      if (closeIcon != null && closeIcon != currentCloseIcon) {
        currentCloseIcon = closeIcon;
        await tester.tap(find.text('Include close Icon'));
      }
      if (multiLineText != null && multiLineText != currentMultiLineText) {
        currentMultiLineText = multiLineText;
        await tester.tap(find.text('Multi Line Text'));
      }
     ...
    }
```

Reply:
I would advice against adding making it complex. How about splitting the tests into different configuration? At the start of the test, select the particular and title each test clearly so this should make it readable.

Reply:
I could re-write your tests but contribution is an excellent way to experiment and learn new techniques.

Reply:
Thanks for the advice, that worked really well.

> contribution is an excellent way to experiment and learn new techniques

I agree and am happy about any feedback I get.